test/api/pack/aggregate_by_test.cpp

```diff
@@ -36,6 +36,12 @@ TEST(sort_by_test, small_example)
     }
 }
 
+TEST(aggregate_by_test, filenames_are_emprty)
+{
+    pack_data data;
+    aggregate_by(data, 0);
+}
```

Comment:
It shouldn't change anything, right?
So maybe:

```suggestion
{
    pack_data data{};

    char original_binary[sizeof(data)];
    std::memcpy(original_binary, &data, sizeof(data));
    
    aggregate_by(data, 0);
    
    char new_binary[sizeof(data)];
    std::memcpy(new_binary, &data, sizeof(data));

    EXPECT_TRUE(std::strcmp(original_binary, new_binary) == 0);
}
```

Or implement the comparison operator :D

Reply:
Yes it does change data. Aggregation means I change the data rows by merging them.
E.g. aggregate by column 2 (specs)
```
file1 spec1 100
file2 spec1 50
file3 spec2 60
``` 
yields
```
file1;file2 spec1 150
file3 spec2 60
```

Reply:
But, `data` is empty here, so it does nothing, right?
Shouldn't we confirm that it does nothing?